### PR TITLE
[ovirt-provider-ovn] Introducing a plugin for ovirt-provider-ovn.

### PR DESCRIPTION
**Pull request by leongold**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/plugins/ovirt_provider_ovn.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class OvirtProviderOvn(Plugin, RedHatPlugin):
+    """oVirt OVN Provider
+    """
+
+    packages = ('ovirt-provider-ovn',)
+    plugin_name = 'ovirt_provider_ovn'
+    profiles = ('virt',)
+
+    def setup(self):
+        self.add_copy_spec(['/etc/ovirt-provider-ovn/ovirt-provider-ovn.conf'])
```
Comment (bmr-cymru) [inline sos/plugins/ovirt_provider_ovn.py]: There's no need for a list here: `Plugin.add_copy_spec()` accepts either a single string argument, or a list of strings.

File: sos/plugins/ovirt_provider_ovn.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        spec = '/var/log/ovirt-provider-ovn.log'
+        if self.get_option('all_logs'):
+            spec += '*'
+        self.add_copy_spec([spec])
```
Comment (bmr-cymru) [inline sos/plugins/ovirt_provider_ovn.py]: Please implement log size limiting using the global `--log-size` option, or add a comment explaining why log size limiting is not useful for these logs (e.g. they are guaranteed to be a fixed or small size).

See for the `openvswitch` plugin for an example of how to do this in combination with `--all-logs`.

File: sos/plugins/ovirt_provider_ovn.py [hunk 3]
```diff
@@ -25,9 +25,10 @@ class OvirtProviderOvn(Plugin, RedHatPlugin):
     profiles = ('virt',)
 
     def setup(self):
-        self.add_copy_spec(['/etc/ovirt-provider-ovn/ovirt-provider-ovn.conf'])
+        self.add_copy_spec('/etc/ovirt-provider-ovn/ovirt-provider-ovn.conf')
 
+        self.limit = self.get_option('log_size')
```
Comment (bmr-cymru) [inline sos/plugins/ovirt_provider_ovn.py]: There's no need to make this a member of the `OvirtProviderOvn` class: it is assigned and referenced in only a single method.

The pattern in the `openvswitch` plugin is perfectly correct.

**Comment by leongold** [comment]
I guess we don't even need a local variable for sizelimit.

**Comment by bmr-cymru** [comment]
Right: for this case, since we're only using it for a single call you can just grab it straight from `get_option()`. There are a few examples in the tree still that get this wrong: we're slowly cleaning those up as we go.

**Comment by leongold** [comment]
Hey Bryn, anything else I'm missing?

**Comment by bmr-cymru** [comment]
Looks good, thanks for making the changes. The only other thing that could come up would be if there could be any sensitive data (keys, passwords or other secrets) in the configuration locations collected - if not then this is ready for merge.

**Comment by leongold** [comment]
Thanks Bryn, this can be merged

**Comment by pmoravec** [comment]
Reading https://github.com/oVirt/ovirt-provider-ovn/blob/master/README.adoc , shouldnt we obfuscate value of e.g. `ovirt-sso-client-secret` ?

Otherwise +1.

**Comment by pmoravec** [comment]
+1

**Comment by leongold** [comment]
ping

**Comment by dougsland** [comment]
+1

**Pull request closed without merging**